benchmark/src/lib.rs

```diff
@@ -478,8 +499,10 @@ impl Benchmarker {
         &mut self,
         txn_reqs: &[SubmitTransactionRequest],
         senders: &mut [AccountData],
+        pattern: &mut Option<SubmitPattern>,
     ) -> (usize, usize, u128, u128) {
-        let (num_txns_accepted, submit_duration_ms) = self.submit_txns(txn_reqs);
+        let submit_pattern = pattern.get_or_insert(self.pattern);
```

Comment:
You're using `get_or_insert` to implement `if`, and for this, you're passing `ref mut None`

Reply:
Updated. `ref mut` is actually not needed here.